resources/repository.rb

```diff
@@ -0,0 +1,88 @@
+property :enable_pgdg_source, [true,false], default: false
+property :enable_pgdg_updates_testing, [true,false], default: true
+property :enable_pgdg_source_updates_testing, [true,false], default: false
+
+default_action :add
+
+action :add do
+  case node['platform_family']
+
+  when 'rhel'
+    remote_file "/etc/pki/rpm-gpg/RPM-GPG-KEY-PGDG-#{new_resource.version}" do
+      source "https://download.postgresql.org/pub/repos/yum/RPM-GPG-KEY-PGDG-#{new_resource.version}"
+    end
+
+    yum_repository "PostgreSQL #{new_resource.version} $releasever - $basearch" do
+      repositoryid "pgdg#{new_resource.version}"
+       baseurl     "https://download.postgresql.org/pub/repos/yum/#{new_resource.version}/redhat/rhel-$releasever-$basearch"
+       enabled     "#{new_resource.enable_pgdg}"
+       gpgcheck    true
+       gpgkey      "file:///etc/pki/rpm-gpg/RPM-GPG-KEY-PGDG-#{new_resource.version}"
+    end
+
+    yum_repository "PostgreSQL #{new_resource.version} $releasever - $basearch - source " do
+      repositoryid "pgdg#{new_resource.version}-source"
+       baseurl     "https://download.postgresql.org/pub/repos/yum/srpms/#{new_resource.version}/redhat/rhel-$releasever-$basearch"
+       enabled     "#{new_resource.enable_pgdg_source}"
+       gpgcheck    true
+       gpgkey      "file:///etc/pki/rpm-gpg/RPM-GPG-KEY-PGDG-#{new_resource.version}"
+    end
+
+    yum_repository "PostgreSQL #{new_resource.version} $releasever - $basearch - updates testing" do
+      repositoryid "pgdg#{new_resource.version}-updates-testing"
+       baseurl     "https://download.postgresql.org/pub/repos/yum/srpms/#{new_resource.version}/redhat/rhel-$releasever-$basearch"
+       enabled     "#{new_resource.enable_pgdg_updates_testing}"
+       gpgcheck    true
+       gpgkey      "file:///etc/pki/rpm-gpg/RPM-GPG-KEY-PGDG-#{new_resource.version}"
+    end
+
+    yum_repository "PostgreSQL #{new_resource.version} $releasever - $basearch - source - updates testing" do
+      repositoryid "pgdg#{new_resource.version}-source-updates-testing"
+       baseurl     "https://download.postgresql.org/pub/repos/yum/srpms/testing/#{new_resource.version}/redhat/rhel-$releasever-$basearch"
+       enabled     "#{new_resource.enable_pgdg_source_updates_testing}"
+       gpgcheck    true
+       gpgkey      "file:///etc/pki/rpm-gpg/RPM-GPG-KEY-PGDG-#{new_resource.version}"
+    end
+
+  when 'debian'
+    remote_file "#{Chef::Config[:file_cache_path]}/ACCC4CF8.asc" do
+      source "https://www.postgresql.org/media/keys/ACCC4CF8.asc"
+      notifies :run, 'bash[apt-key-add]', :immediately
+    end
+
+    bash 'apt-key-add' do
+      code "sudo apt-key add #{Chef::Config[:file_cache_path]}/ACCC4CF8.asc"
+      action :nothing
+    end
+
+    apt_repository 'name' do
+     uri          "http://apt.postgresql.org/pub/repos/apt/"
+     components   ["main"]
+     distribution "#{node['lsb']['codename']}-pgdg"
+     cache_rebuild true
+    end
+
+    apt_update 'update'
```

Comment:
This shouldn't be necessary as the apt_update forces a cache update on the repo

Reply:
Awesome, ta.